### ISSUE TITLE
tendermint rpc with websocket client is not reconnecting

### ISSUE THREAD

**Issue by KrishnaPG** [iov-one/iov-core#1342]
When RPC Client is created using websocket, the connection is not getting reestablished. Instead we see this below error from server:
````
Failed to read request   module=rpc-server protocol=websocket remote=127.0.0.1:63688 
err=\"read tcp 127.0.0.1:26657->127.0.0.1:63688: i/o timeout\"

Error closing connection module=rpc-server protocol=websocket remote=127.0.0.1:63688 
err=\"close tcp 127.0.0.1:26657->127.0.0.1:63688: use of closed network connection\"\n"
````

The problem with this is, if for any reason the connection drops between the rpc client and the tendermint node, then any further RPC calls to the server are not returning any results.

It would be great if the connection is retried whenever it drops.

**Comment by webmaster128** [comment]
Welcome @KrishnaPG and thanks for raising this issue. Which version of @iov/tendermint-rpc are you using? We are already working on reconnecting websocket connections automatically, which turns out to be quite hard.

In https://github.com/iov-one/iov-core/pull/1093 we added a reconnecting websocket that should in theory re-connect every time the connection drops. However, this [does not re-subscribe to subscriptions](https://github.com/iov-one/iov-core/issues/1200).

If you could describe in more details what you try to do (preferribly with caller code) and what exactly does not work, we can try to understand the specific problem better.

**Comment by KrishnaPG** [comment]
Thank you @webmaster128 

The version of @iov/tendermint-rpc used for this is: 1.2.0 (from npm)

I just put some debug breakpoints inside VS to step over the code and verify how to send the parameters correctly to the `broadcastTxSync`. The usage in this case was simple, something like below:
````
const { Client: TMintRPC } = require('@iov/tendermint-rpc');
this.tMintRPCClient = await TMintRPC.connect("ws://127.0.0.1:26657);

proposeBlock(data) {
    return this.tMintRPCClient.broadcastTxSync({ tx: data });
}
````
This works fine with `http`, but with `ws` even few seconds delay in stepping through the code just disconnects and does not reconnect afterwards for next call (even breakpoints are disabled for the next calls).

The culprit is in my opioin, the server side (tendermint node) is disconnecting the client and for some reason client is not able to detect it (which is evident by the fact that client is able to make the request, but server is unable to reply back). The server is what is throwing the error, not the RPCClient.

**Comment by willclarktech** [comment]
Thanks @KrishnaPG, would your issue be solved if #1200 was successfully closed?

**Comment by KrishnaPG** [comment]
On the surface it seems to @willclarktech . Closing this for now.
